test/xbgpu/test_engine.py

```diff
@@ -347,41 +347,98 @@ def verify_corrprod_sensors(
 
 def verify_beam_sensors(
     beam_outputs: list[BOutput],
+    beam_results_shape: tuple[int, ...],
+    prom_diff: PromDiff,
+    actual_sensor_updates: dict[str, list[tuple[Any, aiokatcp.Sensor.Status]]],
     first_timestamp: int,
     last_timestamp: int,
-    actual_sensor_updates: dict[str, list[tuple[Any, aiokatcp.Sensor.Status]]],
     weights: np.ndarray,
     quant_gains: np.ndarray,
     delays: np.ndarray,
 ) -> None:
-    """Verify katcp sensors for BPipeline data.
+    """Verify katcp sensors and Prometheus counters for BPipeline data.
+
+    NOTE: There is currently logic in place to compensate for the BPipeline's
+    lack of tracking missing data. More specifically, it compensates for the
+    difference in expected versus measured counter values by seemingly
+    'forcing' them to match. In actuality, it's counteracting the BPipeline's
+    BSend transmitting `HEAPS_PER_FENGINE_PER_CHUNK` heaps regardless of how
+    empty/full the incoming Chunk is.
 
     Parameters
     ----------
     beam_outputs
         Output beam configurations parsed into BOutput objects.
-    first_timestamp, last_timestamp
-        Two timestamps indicating the start and end of data processing
-        by the :class:`BPipeline`.
+    beam_results_shape
+        The shape of the verified beam data for all beams with the
+        following shape
+        - (len(beam_outputs), n_beam_heaps_sent, n_channels_per_substream,
```

Comment:
Is the `-` intended to be a list bullet? I don't think it'll render that way without more whitespace.